inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+
+# 1. Linearity
+plot(model, which = 1)
+
+# 2. Normality of residuals
+plot(model, which = 2)
+
+# 3. Homoscedasticity
+plot(model, which = 3)
+
+# 4. Independence of residuals (Durbin-Watson test)
+dwtest(model)
+```
+Here, 
+1. Linearity plot(Plot 1) shows the residuals vs. fitted values. The residuals
+should be randomly scattered around zero, indicating that the linearity
+assumption is met. 
+2. Normality/QQ plot(Plot 2) of the residuals should show that the residuals
+are approximately normally distributed. Deviations from the line suggest 
+deviations from normality.
+3. Homoscedasticity (Plot 3): The residuals vs. fitted values plot should
+show constant variance of the residuals (homoscedasticity). If there is a
+pattern (e.g., funnel shape), it indicates heteroscedasticity.
+4. Durbin-Watson Test checks for autocorrelation in the residuals. A value
+close to 2 indicates no autocorrelation. Values significantly different 
+from 2 suggest autocorrelation.
+
+Further, treat sex as another possible covariate :
+
+```{r}
+#| label: sub-trying-possible-covariates
+
+sex <- colData(tse)$SampleType
+
+# Add sex to the data frame
+data$Sex <- sex
+
+# Perform the regression analysis with covariates
+model_with_covariates <- lm(Shannon_Index ~ BMI_Group + Sex, data = data)
+
+# Print the summary of the model with covariates
+summary(model_with_covariates)
+```
+
+Additionally, do visualizations using `scater`.
+```{r}
+#| label: sub-scater-visualization
+
+# Plot PCA for Shannon index
+sce <- SingleCellExperiment(assays = list(counts = assay(tse, "counts")),
+colData = colData(tse))
+
+# Perform PCA and store the results
+sce <- runPCA(sce, ncomponents = 2, exprs_values = "counts")
+
+# Plot PCA with Shannon Index colored by BMI Group
+plotPCA(sce, shape_by = "SampleType") +
+  labs(title = 'PCA of Shannon Index colored by BMI Group')
+```
+The PCA plot shows the first two principal components. Each point represents a
+sample, colored by the Shannon index and shaped by the BMI group. Clusters or 
+separation of points may indicate that certain BMI groups have distinct microbial
+compositions. Overlapping points suggest similar compositions.
+
+Let's also visualize the relationship between Shannon Index and BMI Group
+using a violin plot. Violin plot combines features of a boxplot and a 
+density plot. It shows the distribution and density of the Shannon index for
+each BMI group. Wider sections indicate higher density of data points.
+
+```{r}
+#| label: sub-violin_plot-visualization
+
+violin_plot <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index,
+fill = BMI_Group)) +
+  geom_violin() +
+  geom_jitter(width = 0.2) +
+  labs(title = 'Violin Plot of Shannon Index vs. BMI Group', x = 'BMI Group',
+  y = 'Shannon Index') +
+  theme_minimal()
+
+# Print the violin plot
+print(violin_plot)
+
+# Visualize the regression results with a summary plot
+par(mfrow=c(2,2))  # Setting up a 2x2 plot grid
+plot(model_with_covariates)
+```
```

Comment:
You have violin plots readily available for tse objects in miaViz and should use those. Ask Muluh if you don't find otherwise.